src/span.js

```diff
@@ -0,0 +1,163 @@
+// @flow
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import assert from 'assert';
+import * as thrift from './thrift.js';
+import Utils from './util.js';
+import SpanContext from './span_context.js';
+
+export default class Span {
+    _tracer: any;
+    _name: string;
+    _spanContext: SpanContext;
+    _start: number;
+    _duration: number;
+    _port: number;
+    _isClient: boolean;
+    _peer: Endpoint;
+    _ended: boolean;
+    _annotations: Array<Annotation>;
+    _binaryAnnotations: Array<BinaryAnnotation>;
+
+    constructor(tracer: any,
+                name: string,
+                spanContext: SpanContext,
+                start: number,
+                isClient: boolean = false,
+                peer: Endpoint,
+                annotations: Array<Annotation> = [],
+                binaryAnnotations: Array<BinaryAnnotation> = []
+    ) {
+        this._tracer = tracer;
+        this._name = name;
+        this._spanContext = spanContext;
+        this._start = start;
+        this._duration = 0;
```

Comment:
Undefined makes more sense to me because at this point in time you do not know what duration of the span is, hence it is `undefined`. 0 is not undefined, it's an actual value. In Go it would be natural to use 0 because there's no other choice, but here `undefined` expresses the meaning better.